torch/distributions/exp_family.py

```diff
@@ -0,0 +1,80 @@
+import torch
+from torch.distributions.distribution import Distribution
+from torch.autograd import Variable
+
+
+class ExponentialFamily(Distribution):
+    r"""
+    ExponentialFamily is the abstract base class for probability distributions belonging to an
+    exponential family, whose probability mass/density function has the form is defined below
+
+    ..math::
+
+        p_{F}(x; \theta) = \exp(\langle t(x), \theta\rangle) - F(\theta) + k(x))
+
+    where :math:`\theta` denotes the natural parameters, :math:`t(x)` denotes the sufficient statistic,
+    :math:`F(\theta)` is the log normalizer function for a given family and :math:`k(x)` is the carrier
+    measure.
+
+    Note:
+        This class is an intermediary between the `Distribution` class and distributions which belong
+        to an exponential family mainly to check the correctness of the `.entropy()` and analytic KL
+        divergence methods. We use this class to compute the entropy and KL divergence using the AD frame-
+        work and Bregman divergences (courtesy of: Frank Nielsen and Richard Nock, Entropies and
+        Cross-entropies of Exponential Families).
+    """
+
+    @property
+    def natural_params(self):
+        """
+        Abstract method for natural parameters. Returns a tuple of Variables based
+        on the distribution
+        """
+        raise NotImplementedError
+
+    def log_normalizer(self):
```

Comment:
I think this now needs to be a function
```py
class ExponentialFamily(Distribution):
    def log_normalizer(self, *params):
        raise NotImplementedError

class Exponential(ExponentialFamily):
    def log_normalizer(self, x):
        return -torch.log(-x)
```

Reply:
Does the function signature need to be the same? I changed it without changing the signature in `ExponentialFamily` and it still seems to be fine.

Reply:
I think `log_normalizer(self, *natural_params)` is not needed but would clarify how it should be implemented and would serve as documentation for developers.

Reply:
Is it expected that this function has a different signature across distributions? Does the number of arguments match the number of natural params?